Kernel/Syscalls/execve.cpp

```diff
@@ -131,6 +131,9 @@ static KResultOr<FlatPtr> make_userspace_stack_for_main_thread(Region& region, V
 
     // NOTE: The stack needs to be 16-byte aligned.
     new_esp -= new_esp % 16;
+    // GCC assumes that the return address has been pushed to the stack when it enters the function,
+    // so we need to reserve an extra 4 bytes to make GCC's stack alignment calculations work
+    new_esp -= 4;
```

Comment:
How about `sizeof(void*)` to make this compatible with x86_64 as well?

Reply:
I think the ABI alignment restrictions might be different on x86_64, let me check.

Reply:
Also, can you add a `VERIFY` that will catch this going forward?

Reply:
Oh yeah, a VERIFY would probably be good. Will add that shortly.

Reply:
@IdanHo Apparently it does work the same way on x86 (1 (ptr size) below boundary), I'll update it now. Also, we should eventually move this to an arch specific directory.